unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -841,118 +853,47 @@ prettyDefinitionsForHQName ::
   -- | The name, hash, or both, of the definition to display.
   HQ.HashQualified Name ->
   Backend IO DefinitionDisplayResults
-prettyDefinitionsForHQName path mayRoot renderWidth suffixifyBindings rt codebase query = do
-  (shallowRoot, hqLength) <-
-    (lift . Codebase.runTransaction codebase) do
-      shallowRoot <- resolveCausalHashV2 mayRoot
-      hqLength <- Codebase.hashLength
-      pure (shallowRoot, hqLength)
-  (localNamesOnly, unbiasedPPE) <- scopedNamesForBranchHash codebase (Just shallowRoot) path
+prettyDefinitionsForHQName perspective mayRoot renderWidth suffixifyBindings rt codebase perspectiveQuery = do
```

Comment:
Share's version of this function is drifting further away from the local UCM version, so I ended up just splitting it off as @mitchellwrosen suggested;

I pulled out the helpers I can re-use for now, then removed anything Share specific from this version of the function and moved it to `Unison.Server.Share.Definitions.definitionForHQName`

Sorry for the churn